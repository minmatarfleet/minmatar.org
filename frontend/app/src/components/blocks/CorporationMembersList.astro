---
import { i18n } from '@helpers/i18n'
const { t } = i18n(Astro.url)

import type { Alert, CorporationMembers } from '@dtypes/layout_components'

interface Props {
    corporations:   CorporationMembers[];
    alert?:         Alert | false;
    [propName: string]: any;
}

const {
    corporations,
    alert = false,
    ...attributes
} = Astro.props

import CorporationMembersBlock from '@components/blocks/CorporationMembersBlock.astro';
import TextBox from '@components/layout/TextBox.astro';
import Flexblock from '@components/compositions/Flexblock.astro';
---

<div
    id="corporations-members-list"
    class="[ corporations-members-list ]"
    {...attributes}
>
    {corporations.length > 0 &&
        <Flexblock class="[ correct-gaps ]">
            <p x-show="total_members_filtered == 0" style="display: none;">{t('no_members')}</p>

            {corporations.map( (corporation) => 
                <CorporationMembersBlock corporation={corporation} />
            )}
        </Flexblock>
    }

    {corporations.length == 0 &&
        <slot>
            <TextBox>
                <p>{t('no_members')}</p>
            </TextBox>
        </slot>
    }
</div>