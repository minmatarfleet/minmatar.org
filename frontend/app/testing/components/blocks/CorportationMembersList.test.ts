import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test, vi } from "vitest";
import CorporationMembersList from "@components/blocks/CorporationMembersList.astro";

test("CorporationMembersList defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(CorporationMembersList, {
    props: {
      corporations: [],
    },
    slots: {
      default: "slotted chick",
    },
  });

  // expect(result).toMatchSnapshot();
});
