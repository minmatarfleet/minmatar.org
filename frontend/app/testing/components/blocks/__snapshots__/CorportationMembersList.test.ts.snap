// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CorporationMembersList defaults 1`] = `"<div id="corporations-members-list" class="[ corporations-members-list ]">  slotted chick </div>"`;
